packages/cln_plugin/lib/src/cln_plugin_base.dart

```diff
@@ -1,6 +1,47 @@
 // TODO: Put public facing types in this file.
 
-/// Checks if you are awesome. Spoiler: you are.
-class Awesome {
-  bool get isAwesome => true;
+import 'dart:io';
+
+import 'icln_plugin_base.dart';
+
+class Plugin implements ICLNPlugin {
+  // Data members
+  /*
+  ManifestMethod _manifest;
+  List<Interceptor> _preInterceptor;
+  List<Interceptor> _postInterceptor;
+  Map<String, Object> parameters;
+  bool _parametersReady;
+  JsonConverter converter;
+  CLightningPluginConfig configs;
+   */
+
+  Plugin() {
+    // Data member initializations
+  }
+
+  @override
+  void start() {
+    // This function will communicate with the rpc.
+    print("The plugin is started.");
+    try {
+      String? messageSocket;
+      while((messageSocket = stdin.readLineSync()) != null) {
+        // Already checked is stdin is not null, why trim and check again??
+        if(messageSocket!.trim().isEmpty) {
+          continue;
+        }
+        // TODO: Log messageSocket to stdout as a debug message
+        int deep = 0;
+
+        // Dart doesn't have a max 32 bit int, and casting a double would return errors for some reason, hence the hardcode value.
+        final int maxDeep = 1073741824;
```

Comment:
Sorry for that, this is a tricky stuff.

core lightning mark the end of the stream as `\n\n` and java mark the end of the stream as `\n` in this way there is a this deep mark https://github.com/clightning4j/JRPClightning/blob/master/src/main/java/jrpc/clightning/plugins/CLightningPlugin.java#L126

I hope that this in dart it is not present anymore, and we can skip this bad while loop